File: string/130.cpp
#include <bits/stdc++.h>

using namespace std;

constexpr int conversion = 32;

void debug(const unordered_map<char, int> &um) {
    for(const auto& [k, v]: um) {
        cout<< k << " "<< v <<"\n";
    }

}

int numberOfSpecialChars(string word) {
    unordered_map<char, int> um;

    for (const char& ch : word) {
        if (islower(ch)) {
            if (um[ch - conversion] == 0) {
                um[ch] = 1;
            } else {
                um[ch] = 0;
            }
        } else {
            um[ch]++;
        }
    }

    int answer = 0;
    for (const auto& [ch, cont] : um) {
        if (islower(ch) && um[toupper(ch)] != 0)
            answer += cont;
    }

    return answer;
}

int main() {
    string word = "aaAbcBC";
    cout << numberOfSpecialChars(word) << "\n";    

    return 0;
}

File: string/130.py
def numberOfSpecialChars(word: str) -> int:
    lowercase_table = [0] * 26
    uppercase_table = [0] * 26

    for ch in word:
        character_converted = ord(ch)

        if ch.islower():
            if uppercase_table[character_converted - 32 - 65] == 0:
                lowercase_table[character_converted - 97] = 1
            else:
                lowercase_table[character_converted - 97] = 0
        else:
            uppercase_table[character_converted - 65] = 1

    answer = 0
    for index in range(26):
        if uppercase_table[index] != 0:
            answer += lowercase_table[index]

    return answer


cad = "AbBCab"
print(numberOfSpecialChars(cad))
